### ISSUE TITLE
[BUG] Visualizations created via Saved Objects API do not show up in the web.

### ISSUE THREAD

**Issue by dimittal** [opensearch-project/OpenSearch-Dashboards#2911]
**Describe the bug**
Visualizations created via Saved Objects API are not showing in OpenSearch Dashboard in the visualization section but are accessible via API. Similarly, the visualizations created from the web aren't accessible via Saved objects API. 

**To Reproduce**
1. Go to Visualizations
2. Create a new line visualization from your index pattern
3. Go to saved objects and take its ID
4. Make the saved objects GET api with this ID.
5. The API call returns 404 (Not found). 
6. If you do the other way around and make the visualization from API, it doesn't show up in the visualization section. 

**Expected behavior**
Visualisations created via Saved Objects API should be visible in the OpenSearch Dashboard in the Visualization section

**OpenSearch Version**
2.4.0

**Dashboards Version**
2.4.0

**Host/Environment:**
 - Hosted on Amazon Linux 2 (t2.large)
 - Browser - Firefox 105.0.2 (64-bit) on Mac OS 10.15.5

**Comment by joshuarrrr** [comment]
This sounds like maybe a duplicate of https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1526, although it sounds like potentially more of an issue with security. Can you confirm that you're accessing the same tenant in both cases?

**Comment by dimittal** [comment]
Yes. the issue seems related to #1526. I see I only have one tenant in my dashboard and there is no mention of the tenant when using the API. The namespace is default in my saved object API calls.  

Following is the GET API call response on the endpoint - `api/saved_objects/visualization/9197b020-699e-11ed-9134-d1226417a6c6`. This is the visualization I had created using the API. 
`{
    "id": "9197b020-699e-11ed-9134-d1226417a6c6",
    "type": "visualization",
    "namespaces": [
        "default"
    ],
    "updated_at": "2022-11-21T13:15:28.674Z",
    "version": "WzQ2LDFd",
    "attributes": {
        "title": "Average driver count returned",
        "visState": "{\"title\":\"Average driver count returned\",\"type\":\"line\",\"aggs\":[{\"id\":\"1\",\"enabled\":true,\"type\":\"avg\",\"params\":{\"field\":\"args.drivers._size\"},\"schema\":\"metric\"},{\"id\":\"2\",\"enabled\":true,\"type\":\"date_histogram\",\"params\":{\"field\":\"@timestamp\",\"timeRange\":{\"from\":\"now-30m\",\"to\":\"now\"},\"useNormalizedOpenSearchInterval\":true,\"scaleMetricValues\":false,\"interval\":\"2m\",\"drop_partials\":false,\"min_doc_count\":1,\"extended_bounds\":{}},\"schema\":\"segment\"}],\"params\":{\"type\":\"line\",\"grid\":{\"categoryLines\":false},\"categoryAxes\":[{\"id\":\"CategoryAxis-1\",\"type\":\"category\",\"position\":\"bottom\",\"show\":true,\"style\":{},\"scale\":{\"type\":\"linear\"},\"labels\":{\"show\":true,\"filter\":true,\"truncate\":100},\"title\":{}}],\"valueAxes\":[{\"id\":\"ValueAxis-1\",\"name\":\"LeftAxis-1\",\"type\":\"value\",\"position\":\"left\",\"show\":true,\"style\":{},\"scale\":{\"type\":\"linear\",\"mode\":\"normal\"},\"labels\":{\"show\":true,\"rotate\":0,\"filter\":false,\"truncate\":100},\"title\":{\"text\":\"Average args.drivers._size\"}}],\"seriesParams\":[{\"show\":true,\"type\":\"line\",\"mode\":\"normal\",\"data\":{\"label\":\"Average args.drivers._size\",\"id\":\"1\"},\"valueAxis\":\"ValueAxis-1\",\"drawLinesBetweenPoints\":true,\"lineWidth\":2,\"interpolate\":\"linear\",\"showCircles\":true}],\"addTooltip\":true,\"addLegend\":true,\"legendPosition\":\"right\",\"times\":[],\"addTimeMarker\":false,\"labels\":{},\"thresholdLine\":{\"show\":false,\"value\":10,\"width\":1,\"style\":\"full\",\"color\":\"#E7664C\"}}}",
        "uiStateJSON": "{}",
        "description": "",
        "version": 1,
        "kibanaSavedObjectMeta": {
            "searchSourceJSON": "{\"query\":{\"query\":\"\",\"language\":\"kuery\"},\"filter\":[],\"indexRefName\":\"kibanaSavedObjectMeta.searchSourceJSON.index\"}"
        }
    },
    "references": [
        {
            "name": "kibanaSavedObjectMeta.searchSourceJSON.index",
            "type": "index-pattern",
            "id": "1a1b1d80-699d-11ed-9134-d1226417a6c6"
        }
    ],
    "migrationVersion": {
        "visualization": "7.10.0"
    }
}`

The one I created from web isn't queryable using the APIs.

**Comment by dimittal** [comment]
@joshuarrrr I figured this was a tenancy related issue. I can see my visualizations created via API are going into the private tenant created for the admin user and not showing up when i use the global tenant on the web. Is there a way I can disable multi-tenancy through passing config in my docker compose file? I see there is mention of being able to pass those configs in the compose file here (https://opensearch.org/docs/latest/dashboards/install/docker/) but explicitly mentioning those configs in the environment section isn't working. Any leads would be appreciated.

**Comment by joshuarrrr** [comment]
@dimittal The maintainers  of the security plugin will likely have more expertise to point you in the right direction than I do. @opensearch-project/transfer-request please transfer to https://github.com/opensearch-project/security-dashboards-plugin

**Comment by dtaivpp** [comment]
Ah after looking back at it these are the same issue. Good catch @dimittal  and @joshuarrrr.
